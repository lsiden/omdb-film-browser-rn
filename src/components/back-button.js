import React from "react"
import PropTypes from "prop-types"
import { Icon } from "react-native-elements"
import { TouchableHighlight } from "react-native"

const style = {
  marginLeft: 5,
}
const BackButton = ({ dispatchToPrev, color }) => (
  <TouchableHighlight
    onPress={dispatchToPrev}
    activeOpacity={0.5}
    style={style}
  >
    <Icon name="arrow-back" color={color} />
  </TouchableHighlight>
)

BackButton.propTypes = {
  color: PropTypes.string,
  dispatchToPrev: PropTypes.func.isRequired,
}

BackButton.defaultProps = {
  color: "black",
}

export default BackButton
