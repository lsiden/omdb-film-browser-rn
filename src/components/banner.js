import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import { Icon } from "react-native-elements"
import { TouchableHighlight, Text, View } from "react-native"

import { appStyles as style } from "styles"
import { viewAbout } from "actions"

export const banner = ({ displayInfoIcon, dispatchViewAbout }) => (
  <View style={style.banner}>
    <Text style={style.title}>{"Search Open Movie Database"}&nbsp;</Text>
    {displayInfoIcon && (
      <TouchableHighlight onPress={() => dispatchViewAbout()}>
        <Icon name={"info-outline"} color={"white"} />
      </TouchableHighlight>
    )}
  </View>
)

banner.propTypes = {
  displayInfoIcon: PropTypes.bool,
  dispatchViewAbout: PropTypes.func.isRequired,
}

banner.defaultProps = {
  displayInfoIcon: false,
}

export default connect(null, dispatch => ({
  dispatchViewAbout: () => dispatch(viewAbout()),
}))(banner)
