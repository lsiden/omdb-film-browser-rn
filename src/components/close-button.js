import React from "react"
import PropTypes from "prop-types"

import ButtonLink from "./button-link"

const CLOSE_ICON = "\u2715"

const closeButtonStyle = {
  margin: "8pt",
  fontSize: "20pt",
}

export const CloseButton = ({ onPress }) => (
  <ButtonLink
    onPress={onPress}
    id="close-button"
    title="Close"
    addStyle={closeButtonStyle}
  >
    {CLOSE_ICON}
  </ButtonLink>
)

CloseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default CloseButton
