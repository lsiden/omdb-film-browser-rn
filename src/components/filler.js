// Invisible filler to offset back button so that title content can be centered.
import React from "react"
import { Text, View } from "react-native"

const fillerStyle = { width: 24 }
const fillerTextStyle = { color: "transparent" }

const Filler = () => (
  <View style={fillerStyle}>
    <Text style={fillerTextStyle}>x</Text>
  </View>
)

export default Filler
