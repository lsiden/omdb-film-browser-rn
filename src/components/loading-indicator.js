import React from "react"
import { ActivityIndicator, View } from "react-native"

const blue = "#00f"

const wrapperStyle = {
  justifyContent: "center",
  height: "100%",
}
const LoadingIndicator = () => (
  <View style={wrapperStyle}>
    <ActivityIndicator size="large" color={blue} />
  </View>
)

export default LoadingIndicator
