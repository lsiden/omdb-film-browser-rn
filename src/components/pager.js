import React from "react"
import { connect } from "react-redux"
import PropTypes from "prop-types"
import { View, Text } from "react-native"
import { Button } from "react-native-elements"

import { fetchPage } from "actions/fetch"
import { pagerStyles, appStyles } from "styles"

const icon = (name, disabled) => ({
  name,
  type: "material-community",
  size: 25,
  color: disabled ? "gray" : "black",
})

const btnStyle = {
  backgroundColor: "transparent",
  height: 40,
}

export const pager = props => {
  const { pageNum, lastPage, dispatchFetchPage } = props
  const isFirstPage = n => n == 1
  const isLastPage = n => n == lastPage
  return (
    lastPage > 1 && (
      <View style={pagerStyles.wrapper}>
        <View style={pagerStyles.buttons}>
          <Button
            icon={icon("skip-backward", isFirstPage(pageNum))}
            onPress={() => dispatchFetchPage(1)}
            disabled={isFirstPage(pageNum)}
            buttonStyle={btnStyle}
          />
          <Button
            icon={icon("step-backward", isFirstPage(pageNum))}
            onPress={() => dispatchFetchPage(pageNum - 1)}
            disabled={isFirstPage(pageNum)}
            buttonStyle={btnStyle}
          />
          <Button
            icon={icon("step-forward", isLastPage(pageNum))}
            onPress={() => dispatchFetchPage(pageNum + 1)}
            disabled={isLastPage(pageNum)}
            buttonStyle={btnStyle}
          />
          <Button
            icon={icon("skip-forward", isLastPage(pageNum))}
            onPress={() => dispatchFetchPage(lastPage)}
            disabled={isLastPage(pageNum)}
            buttonStyle={btnStyle}
          />
        </View>
        <Text style={[appStyles.message, pagerStyles.message]}>
          {`On page ${pageNum.toLocaleString()} of ${lastPage.toLocaleString()}`}
        </Text>
      </View>
    )
  )
}

pager.propTypes = {
  pageNum: PropTypes.number.isRequired,
  totalResults: PropTypes.number.isRequired,
  lastPage: PropTypes.number.isRequired,
  dispatchFetchPage: PropTypes.func.isRequired,
}

export default connect(
  state => ({
    pageNum: state.pageNum,
    totalResults: state.totalResults,
    lastPage: state.lastPage,
  }),
  dispatch => ({
    dispatchFetchPage: pageNum => dispatch(fetchPage(pageNum)),
  })
)(pager)
