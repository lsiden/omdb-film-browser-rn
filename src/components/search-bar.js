// Debounces react-native-elements.SearchBar

import React from "react"
import { connect } from "react-redux"
import PropTypes from "prop-types"
import { SearchBar } from "react-native-elements"
import { Platform } from "react-native"

import { fetchQuery } from "actions/fetch"

export class searchBar extends React.Component {
  static propTypes = {
    dispatchQuery: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props)
    this.state = { query: "" }
    this.onChangeText = this.onChangeText.bind(this)
    this.onTimeout = this.onTimeout.bind(this)
    this.onCancel = this.onCancel.bind(this)
  }

  onTimeout() {
    const { query } = this.state
    this.timeoutId = null

    if (query) {
      this.props.dispatchQuery(query)
    }
  }

  onChangeText(query) {
    this.setState({ query })

    if (this.timeoutId) {
      clearTimeout(this.timeoutId)
    }
    this.timeoutId = setTimeout(this.onTimeout, this.props.delay)
  }

  onCancel() {
    this.setState({ query: "" })
    this.ref.clear()
  }

  render() {
    return (
      <SearchBar
        placeholder="Title..."
        onChangeText={this.onChangeText}
        onCancel={this.onCancel}
        clearIcon={{ color: "#86939e", name: "close" }}
        platform={Platform.OS}
      />
    )
  }
}

searchBar.propTypes = {
  dispatchQuery: PropTypes.func.isRequired,
  delay: PropTypes.number
}

searchBar.defaultProps = {
  delay: 300
}

export default connect(null, dispatch => ({
  dispatchQuery: query => {
    dispatch(fetchQuery(query))
  }
}))(searchBar)
