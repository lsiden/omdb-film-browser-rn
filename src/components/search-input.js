import React from "react"
import PropTypes from "prop-types"
import { View, Text, TextInput } from "react-native"

export const SEARCH_ICON = "\u26b2"

const wrapperStyle = {
  position: "relative",
  flex: "1 1",
}

const inputStyle = {
  height: 24,
  width: "95%",
  marginLeft: 24,
  marginTop: 4,
  fontSize: 12,
}

const iconStyle = {
  position: "absolute",
  top: 3,
  left: 6,
  color: "#888",
  fontSize: 20,
  transform: [{ rotate: "-45deg" }],
}

export const SearchInput = props => {
  const { addStyle, ...passProps } = props
  const style = { ...wrapperStyle, ...addStyle }
  return (
    <view style={style}>
      <Text
        style={iconStyle}
        charSet="utf-8"
        dangerouslySetInnerHTML={{ __html: SEARCH_ICON }}
      />
      <TextInput style={inputStyle} {...passProps} />
    </view>
  )
}

SearchInput.propTypes = {
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  addStyle: PropTypes.object,
}

SearchInput.defaultProps = {
  placeholder: "",
  addStyle: {},
}

export default SearchInput
