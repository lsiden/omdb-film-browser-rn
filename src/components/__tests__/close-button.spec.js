import React from "react"
import renderer from "react-native-test-utils"

import { CloseButton } from "components/close-button"

let onPress

const defaultProps = () => {
  onPress = jest.fn()
  return {
    onPress,
  }
}

const createWrapper = (props = {}) =>
  renderer(<CloseButton {...{ ...defaultProps(), ...props }} />)

test("renders()", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
