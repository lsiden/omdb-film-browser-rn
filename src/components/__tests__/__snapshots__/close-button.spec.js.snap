// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders() 1`] = `
<button
  id="close-button"
  onPress={[MockFunction]}
  style={
    Object {
      "background": "none",
      "border": "none",
      "cursor": "pointer",
      "fontSize": 20,
      "margin": 8,
      "padding": "0!important",
    }
  }
  title="Close"
>
  âœ•
</button>
`;
